ocaml/tests/test_nm.ml

```diff
@@ -0,0 +1,72 @@
+open Test_highlevel
+open Test_common
+open OUnit
+
+module MaybeUpdateMasterPifMac = Generic.Make(Generic.EncapsulateState(struct
+  module Io = struct
+    type input_t = (bool * string option) (* auto_update_mac, new_primary_slave_mac *)
+
+    type output_t = (string * string)  (* record bond master mac, db bond master mac *)
+
+    let show_auto_update_mac = fun auto_update_mac -> "auto_update_mac="^string_of_bool auto_update_mac
+    let show_primary_slave_mac_change = fun o_new_primary_slave_mac -> match o_new_primary_slave_mac with
+      | None -> "primary_slave mac unchanged"
+      | Some new_primary_slave_mac -> "primary_slave mac -> "^new_primary_slave_mac
+
+    let string_of_input_t = Test_printers.pair show_auto_update_mac show_primary_slave_mac_change
+    let string_of_output_t (rc,db) = "Bond master macs. record: "^rc^" "^"db : "^db
+  end
+
+  module State  = struct
```

Comment:
You could also do:
```
module State = Test_state.XapiDb
```
See test_state.ml. This is what we tend to use for unit tests that need a xapi DB.

Reply:
I did try that, but i want to carry around refs to the bond and the pif master as well